server/game/KeywordAbilities/RallyAbility.js

```diff
@@ -0,0 +1,39 @@
+const TriggeredAbility = require('../triggeredability.js');
+const { AbilityTypes, Locations } = require('../Constants');
+
+class RallyAbility extends TriggeredAbility {
+    constructor(game, card) {
+        super(game, card, AbilityTypes.ForcedReaction, {
+            when: {
+                onCardRevealed: (event, context) => event.card === context.source &&
+                    [Locations.StrongholdProvince, Locations.ProvinceOne, Locations.ProvinceTwo, Locations.ProvinceThree, Locations.ProvinceFour].includes(event.card.location) &&
+                    context.source.hasRally()
+            },
+            location: [Locations.StrongholdProvince, Locations.ProvinceOne, Locations.ProvinceTwo, Locations.ProvinceThree, Locations.ProvinceFour],
+            title: card.name + '\'s Rally',
+            printedAbility: false,
```

Comment:
I'm not 100% on this, but don't we need to actually check whether Rally is printed on the card and note that here? copying printed abilities from a card in a province is unlikely, but still..

Reply:
Not sure what you mean about this one.

Reply:
My mistake. Pride etc are implemented the same way .